Update to clang format 12.0.1

SimeonEhrig opened the pull request.

psychocoderHPC commented on .clang-format:
```diff
@@ -91,26 +99,20 @@ TabWidth: 4
 UseTab: Never
 
-# Project specific options
-IncludeCategories:
```
Do not remove the include.

SimeonEhrig committed:
Updated to clang-format 12

- update clang-format CI job to clang-format version 12.0.1
- add latest version of the .clang-format file from
 ComputationalRadiationPhysics/contributing (f4c981deb67d0bd1ec8e2d9712dee5608708abe0)
- removed `IndentPragmas` from `.clang-format`, because it was not supported anymore

Commit:
apply clang-format 12 on source code

SimeonEhrig commented:
@psychocoderHPC Can you please review again.
